src/Adapter/Product/QueryHandler/GetProductForEditingHandler.php

```diff
@@ -136,12 +146,34 @@ public function handle(GetProductForEditing $query): ProductForEditing
             $this->getDetails($product),
             $this->getShippingInformation($product),
             $this->getSeoOptions($product),
-            $product->getAssociatedAttachmentIds(),
+            $this->getAttachments($query->getProductId()),
             $this->getProductStockInformation($product),
             $this->getVirtualProductFile($product)
         );
     }
 
+    /**
+     * @param ProductId $productId
+     *
+     * @return ProductAttachment[]
+     */
+    private function getAttachments(ProductId $productId): array
```

Comment:
Please add unit test

Reply:
Confirmed: let's try to verify the output is correct (although this is a `private` method so let's rather test the `public` method 😄 )

Reply:
we've never tested cqrs handlers with units. That is what we use behats for. 
So I'll update the behat scenarios to check not only associated ids, but also other fields (the ones present in query result dto).
Btw, chill, its WIP :sweat_smile:

Reply:
Oops misread it was in a query headnler...


Please add behat tests :trollface: